### ISSUE TITLE
SSLFactory withTrustMaterial with null (empty password)

### ISSUE THREAD

**Issue by luismospinam** [Hakky54/sslcontext-kickstart#11]
Hello.

I'm trying to load my own TrustMaterial which i have in my own JKS keystore.
But I see you have a validations in all the overloaded methods **withTrustMaterial** that the password can not be NULL or empty char[].
Do i really need to add a password to my keystore? Right now my JKS has no password at all.

```  
SSLFactory sslFactory = SSLFactory.builder()
        .withDefaultTrustMaterial()
        .withTrustMaterial("keystore/custom-keystore.jks", null)
        .build();
```

**Comment by Hakky54** [comment]
Hi @luismospinam 

Thank you for stopping by and mentioning the limitation of the library. I initially added those validations as an indicator for the developer that it is not safe to use a non password protected keystores. But maybe the library is too strict in this case and it shouldn't prevent loading the file while there is no password at all. 

I created a pull request with example unit test, loading trust material without password: https://github.com/Hakky54/sslcontext-kickstart/pull/12/files#diff-f2141e4e33c15f6d97b50ba248c5b679R80 and loading identity material without password: https://github.com/Hakky54/sslcontext-kickstart/pull/12/files#diff-f2141e4e33c15f6d97b50ba248c5b679R240. Let me know if this covers your use case

**Comment by luismospinam** [comment]
Thanks for the quick reply @Hakky54 .
In my case I just need to create an **SslContext** from a keystore with just one Public certificate that is invalid to the JVM for some reason (even when the certificate is correct and valid). So there's not much to secure there.
I think that PR covers my use case.
I appreciate your help and thanks for creating this library.

**Comment by Hakky54** [comment]
Great to hear! I merged it and made a release. These changes are now available at version 4.1.0
